client/app/components/Select.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from "react";
+import { maxBy } from "lodash";
+import AntdSelect, { SelectProps } from "antd/lib/select";
+import { calculateTextWidth } from "@/lib/calculateTextWidth";
+
+function Select({ options, ...props }: SelectProps<any>): JSX.Element {
+  const [dropdownMatchSelectWidth, setDropdownMatchSelectWidth] = useState<number | boolean>(true);
+  useEffect(() => {
+    if (options && options.length > 400) {
```

Comment:
Let's move this `400` to a `const MIN_LENGTH_FOR_VIRTUAL_SCROLL` (or any better naming) below the imports to have the use clear and it work as a "definition" for the component.